scripts/lib/authority-and-program-updater.ts

```diff
@@ -90,6 +102,18 @@ type AuthorityMap = {
   [index: AuthorityKey]: Authority;
 };
 
+// These types encapsulate how we parse and then re-write JSON to the data/authorities.json file.
+type JsonAuthorities = {
+  [index: AuthorityKey]: { name: string };
+};
+type AuthorityType = string;
+type AuthorityTypeMap = {
+  [index: AuthorityType]: JsonAuthorities;
+};
+export type StateToAuthorityTypeMap = {
+  [index: StateKey]: AuthorityTypeMap;
+};
```

Comment:
Does the `AuthoritiesByState` type in `src/data/authorities.ts` work here instead?

If not, I'm guessing the `AuthoritiesByType` type I made won't help either, but that's also there.

Maybe there's some gotcha but those types are basically these types, so it would be nice to avoid duplicate definitions if possible.

Reply:
Yeah, I experimented extensively with both types, trying to make them fit here. The major issues I've seen are due to us going through the object entries in a fairly unstructured way: e.g. json[stateUpper][authority.authority_type.toLowerCase()] doesn't have any explicit key value for what that authority type should be, so there are many errors thrown unless we do a switch case for each type of possible authority type there (and then, do the same kind of enumeration for API_AUTHORITY_SCHEMA properties as we're writing and creating). 

There's also the issue of possible "undefined" properties when doing it this way for both `AuthoritiesByState` and `AuthoritiesByType`. 

While I do agree reusing the actual types would be extremely beneficial to maintainability, I'd rather do it in a separate change, since it might change the way things are written/defined here altogether. Does that sound reasonable?